File: src/main/java/com/nugurang/service/ImageService.java
package com.nugurang.service;

import com.nugurang.dao.ImageDao;
import com.nugurang.entity.ImageEntity;
import java.util.Optional;
import org.springframework.stereotype.Service;

@Service
public class ImageService {
    private final ImageDao imageDao;

    public ImageEntity createImage(String imageAddress) {
        return imageDao.save(ImageEntity.builder().address(imageAddress).build());
    }

    public Optional<ImageEntity> getImage(Long id) {
        return imageDao.findById(id);
    }

    public Optional<ImageEntity> getImage(String address) {
        return imageDao.findByAddress(address);
    }

    public void deleteImage(Long id) {
        imageDao.deleteById(id);
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public ImageService(final ImageDao imageDao) {
        this.imageDao = imageDao;
    }
    //</editor-fold>
}

File: src/main/java/com/nugurang/service/ImageService.kt
package com.nugurang.service

import com.nugurang.dao.ImageDao
import com.nugurang.entity.ImageEntity
import com.nugurang.exception.NotFoundException
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class ImageService(private val imageDao: ImageDao) {

    fun createImage(imageAddress: String): ImageEntity {
        return imageDao.save(ImageEntity.builder().address(imageAddress).build())
    }

    fun getImage(id: Long): ImageEntity {
        return imageDao.findByIdOrNull(id) ?: throw NotFoundException(ImageEntity::class.java)
    }

    fun getImage(address: String): ImageEntity {
        return imageDao.findByAddress(address) ?: throw NotFoundException(ImageEntity::class.java)
    }

    fun deleteImage(id: Long) {
        imageDao.deleteById(id)
    }
}
